libs/queue/SPECIFICATION.md

```diff
@@ -0,0 +1,35 @@
+# Overview
+
+This document provides an overview of the Queue library.
+
+It outlines the use cases, i.e. specification, and describes how to implement the library.
+
+## Use Cases
+
+The Queue library can be used as a traditional queue or first in first out data structure.
+
+## Public Functions
+
+### `new()`
+
+Creates a new instance of a `Queue`. 
```

Comment:
```suggestion
- Creates a new instance of a `Queue`. 
```

I would add bullet points to all of these to make them more readable (especially when more items are added) and consistent across larger libraries (that probably do this / ought to if they are not).